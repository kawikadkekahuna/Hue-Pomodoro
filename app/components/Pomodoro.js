import glamorous from "glamorous";
import { lightState } from "node-hue-api";
import React, { Component } from "react";
import { Link } from "react-router";
import AppSettingsIcon from "material-ui/svg-icons/action/settings";
import LightSettingsIcon from "material-ui/svg-icons/action/lightbulb-outline";
import Timer from "./Timer";
import Button from "./Button";

const { Div } = glamorous;

export default class Pomodoro extends Component {
  render() {
    return (
      <div>
        <Div position="absolute" top="0" right="0">
          <Link to="/hue-config"><AppSettingsIcon /></Link>
          <Link to="/hue-config"><LightSettingsIcon /></Link>
        </Div>
        <Div>
          <Timer time={this.props.timeTillFinish} />
          <Div display="flex" justifyContent="center" alignItems="center">
            <Button
              className="pomodoroToggle"
              onClick={() => {
                this.props.handlePomodoroEvent();
              }}
              text={this.props.tickState}
            />
            <Button onClick={this.props.resetTimer} text="Reset" />
          </Div>
        </Div>
      </div>
    );
  }
}
