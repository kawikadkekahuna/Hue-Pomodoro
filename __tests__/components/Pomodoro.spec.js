import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Pomodoro from '../../app/components/Pomodoro';

describe('Pomodoro Component', () => {
  test('has correct default state', () => {
    const tree = shallow(<Pomodoro />);
    expect(toJson(tree)).toMatchSnapshot();
  });
  test('can handle pomodoro event', () => {
    const props = {
      handlePomodoroEvent: jest.fn()
    };
    const tree = shallow(<Pomodoro {...props} />);
    tree.find('.pomodoroToggle').simulate('click');
    expect(props.handlePomodoroEvent).toHaveBeenCalled();
    expect(toJson(tree)).toMatchSnapshot();
  });
});
