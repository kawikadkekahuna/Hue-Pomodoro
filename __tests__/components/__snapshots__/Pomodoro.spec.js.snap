// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pomodoro Component can handle pomodoro event 1`] = `
<div>
  <glamorous.Div
    position="absolute"
    right="0"
    top="0"
  >
    <Link
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/hue-config"
    >
      <ActionSettings />
    </Link>
    <Link
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/hue-config"
    >
      <ActionLightbulbOutline />
    </Link>
  </glamorous.Div>
  <glamorous.Div>
    <Timer />
    <glamorous.Div
      alignItems="center"
      display="flex"
      justifyContent="center"
    >
      <Button
        className="pomodoroToggle"
        onClick={[Function]}
      />
      <Button
        text="Reset"
      />
    </glamorous.Div>
  </glamorous.Div>
</div>
`;

exports[`Pomodoro Component has correct default state 1`] = `
<div>
  <glamorous.Div
    position="absolute"
    right="0"
    top="0"
  >
    <Link
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/hue-config"
    >
      <ActionSettings />
    </Link>
    <Link
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/hue-config"
    >
      <ActionLightbulbOutline />
    </Link>
  </glamorous.Div>
  <glamorous.Div>
    <Timer />
    <glamorous.Div
      alignItems="center"
      display="flex"
      justifyContent="center"
    >
      <Button
        className="pomodoroToggle"
        onClick={[Function]}
      />
      <Button
        text="Reset"
      />
    </glamorous.Div>
  </glamorous.Div>
</div>
`;
